### DISCUSSION TITLE
Setting up a data-plane cluster across two vsphare data centers

### DISCUSSION THREAD

**Discussion by sina370** [aws/eks-anywhere#8553] [Q&A]
Having two data centers (two separate cluster - each have 4 esxi) within Vsphare which they share storage with stretch cluster. Is it possible to set up a single EKS-A data-plane cluster across both data centers ? or we have to have two data-plane clusters, one for each data center ?

**Comment by uldyssian-sh** [comment]
With EKS Anywhere on vSphere, a **single Kubernetes cluster (including its data plane / worker nodes)** is designed to live within **one vSphere datacenter** as defined in the VSphereDatacenterConfig.

You cannot span a single EKS Anywhere data-plane cluster across two separate vSphere datacenters, even if:
- the ESXi clusters share stretched storage
- networking is routable between sites
- latency is low

Why this is not supported:

EKS Anywhere’s vSphere provider assumes:
- one vSphere Datacenter object
- one set of vSphere inventory paths (networks, datastores, resource pools)
- homogeneous failure domains within that datacenter

vSphere “datacenter” is a hard boundary in the provider logic. Cross-datacenter placement is not modeled as an availability or failure domain, and EKS-A does not have awareness of stretched clusters or metro-vSphere layouts.

What *is* supported:

1) Multiple EKS Anywhere clusters  
You can create:
- one EKS-A cluster per vSphere datacenter  
Each cluster has its own control plane and data plane.

2) Application-level multi-cluster strategy  
If your goal is resiliency across sites:
- run identical workloads in both clusters
- use DNS, global load balancing, or application-level replication
- optionally use GitOps to keep them in sync

3) Control-plane HA *within* a single datacenter  
Within one vSphere datacenter, you can:
- spread control-plane nodes across ESXi hosts
- spread worker nodes across ESXi hosts
But still within that single datacenter boundary.

What is *not* supported:

- One EKS Anywhere cluster with worker nodes split across two vSphere datacenters
- A single Kubernetes control plane stretched across vSphere datacenters
- Treating stretched storage as a supported multi-DC failure domain

Summary:

A single EKS Anywhere data-plane cluster cannot span two vSphere datacenters.
You must deploy one EKS-A cluster per datacenter.
For multi-DC resilience, use multiple clusters and handle failover at the application or traffic-routing layer.

**Reply by uldyssian-sh** [reply]
Thanks! If this helps, please consider marking it as “Helpful” or “Accepted” 😊
